centreon-certified/influxdb/influxdb-metrics-apiv2.lua

```diff
@@ -0,0 +1,448 @@
+
+  local http_response_body = ""
+  local http_request = curl.easy()
+                           :setopt_url(url)
+                           :setopt_writefunction(
+    function(response)
+      http_response_body = http_response_body .. tostring(response)
+    end
+  )
+                           :setopt(curl.OPT_TIMEOUT, self.sc_params.params.connection_timeout)
+                           :setopt(curl.OPT_SSL_VERIFYPEER, self.sc_params.params.verify_certificate)
+                           :setopt(curl.OPT_HTTPHEADER, queue_metadata.headers)
+
+  -- set proxy address configuration
+  if (self.sc_params.params.proxy_address ~= '') then
+    if (self.sc_params.params.proxy_port ~= '') then
+      http_request:setopt(curl.OPT_PROXY, self.sc_params.params.proxy_address .. ':' .. self.sc_params.params.proxy_port)
+    else
+      self.sc_logger:error("[EventQueue:send_data]: proxy_port parameter is not set but proxy_address is used")
+    end
+  end
+
+  -- set proxy user configuration
+  if (self.sc_params.params.proxy_username ~= '') then
+    if (self.sc_params.params.proxy_password ~= '') then
+      http_request:setopt(curl.OPT_PROXYUSERPWD, self.sc_params.params.proxy_username .. ':' .. self.sc_params.params.proxy_password)
+    else
+      self.sc_logger:error("[EventQueue:send_data]: proxy_password parameter is not set but proxy_username is used")
+    end
+  end
+
+  -- adding the HTTP POST data
+  http_request:setopt_postfields(data_binary)
+
+  -- performing the HTTP request
+  http_request:perform()
+
+  -- collecting results
+  http_response_code = http_request:getinfo(curl.INFO_RESPONSE_CODE)
+
+  http_request:close()
+
+  -- Handling the return code
+  local retval = false
+  -- https://docs.influxdata.com/influxdb/cloud/api/#operation/PostWrite other than 204 is not good
+  if http_response_code == 204 then
+    self.sc_logger:info("[EventQueue:send_data]: HTTP POST request successful: return code is " .. tostring(http_response_code))
+    retval = true
+  else
+    self.sc_logger:error("[EventQueue:send_data]: HTTP POST request FAILED, return code is " .. tostring(http_response_code) .. ". Message is: " .. tostring(http_response_body))
+
+    if payload then
+      self.sc_logger:error("[EventQueue:send_data]: sent payload was: " .. tostring(payload))
+    end
+  end
+
+  return retval
+end
+
+--------------------------------------------------------------------------------
+-- Required functions for Broker StreamConnector
+--------------------------------------------------------------------------------
+
+local queue
+
+-- Fonction init()
+function init(conf)
+  queue = EventQueue.new(conf)
+end
+
+-- --------------------------------------------------------------------------------
+-- write,
+-- @param {table} event, the event from broker
+-- @return {boolean}
+--------------------------------------------------------------------------------
+function write (event)
+  if event._type == 196617 or event._type == 196609 then
```

Comment:
```suggestion
  if queue.sc_params.params.bbdo.categories["storage"].id == event.category and queue.sc_params.params.bbdo.elements["metric"].id == event.element then
```

we must never use those internal id in stream connectors code.